src/main/java/com/library/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.library.service.impl;
+
+import com.library.exception.MissingRequiredDataException;
+import com.library.model.dto.BookDto;
+import com.library.model.entity.Book;
+import com.library.model.mapper.BookMapper;
+import com.library.repository.BookRepository;
+import com.library.service.BookService;
+import com.library.service.ValidateService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import jakarta.persistence.EntityNotFoundException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class BookServiceImpl implements BookService, ValidateService<Book> {
+
+    private final BookRepository bookRepository;
+
+    private final BookMapper bookMapper;
+
+    @Override
+    public List<BookDto> findAllWithPageable(Integer page, Integer sizeOnPage) {
+        Pageable paging = PageRequest.of(page - 1, sizeOnPage);
+        return bookRepository.findAll(paging).stream()
+                .map(bookMapper::mapToBookDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        this.validateId(id);
+        return bookMapper.mapToBookDto(bookRepository.findById(id).orElseThrow(EntityNotFoundException::new));
+    }
+
+    @Override
+    public BookDto save(BookDto bookDto) {
+        Book savedBook = bookMapper.mapToBook(bookDto);
+        this.validateEntity(savedBook);
+        return bookMapper.mapToBookDto(bookRepository.save(savedBook));
+    }
+
+    @Override
+    public BookDto update(BookDto bookDto) {
+        Book updateBook = bookMapper.mapToBook(bookDto);
+        this.validateId(bookDto.getId());
+        Book existBook = bookRepository.getReferenceById(bookDto.getId());
+        updateBook.setAuthors(existBook.getAuthors());
+        updateBook.setPublisher(existBook.getPublisher());
+        return bookMapper.mapToBookDto(bookRepository.save(updateBook));
```

Comment:
What if on update form was added new author?

Reply:
I already delete this method